aten/src/ATen/native/cuda/BatchLinearAlgebraLib.cu

```diff
@@ -497,27 +489,37 @@ std::tuple<Tensor, Tensor, Tensor> _svd_helper_cuda_lib(const Tensor& self, bool
     _create_U_S_VT(self, some, compute_uv, /* svd_use_cusolver = */ true);
   // U, S, V working copies are already column majored now
 
-  // heuristic for using `gesvdjBatched` over `gesvdj`
-  if (m <= 32 && n <= 32 && batch_size > 1 && (!some || m == n)) {
-    apply_svd_lib_gesvdjBatched(self, U_working_copy, S_working_copy, VT_working_copy, infos, compute_uv);
-  } else {
-    apply_svd_lib_gesvdj(self, U_working_copy, S_working_copy, VT_working_copy, infos, compute_uv, some);
+  // gesvdj computes V instead of V^H
+  if (compute_uv) {
+    VT_working_copy.transpose_(-2, -1);
   }
 
-  // A device-host sync will be performed.
-  batchCheckErrors(infos, "svd_cuda");
-
+  // gesvdjBatched fails with illegal memory access and
+  // gesvdj fails with CUSOLVER_STATUS_EXECUTION_FAILED
+  // if matrices for U and VT are not allocated
+  // even though the result of computation is not used we need to allocate this memory
+  Tensor U_tmp = U_working_copy;
+  Tensor VT_tmp = VT_working_copy;
   if (!compute_uv) {
-    VT_working_copy.zero_();
-    U_working_copy.zero_();
+    auto sizes = self.sizes().vec();
+    auto m = self.size(-2);
+    auto n = self.size(-1);
+    sizes[self.dim() - 1] = m;
+    U_tmp = at::empty(sizes, U_working_copy.options());
+    sizes[self.dim() - 2] = n;
+    sizes[self.dim() - 1] = some ? std::min(m, n) : n;
+    VT_tmp = at::empty(sizes, VT_working_copy.options());
```

Comment:
Is it possible to move this allocation part into `_create_U_S_VT`, and, for example, change the `U_empty` and `VT_empty` ternary operators to `(compute_uv || svd_use_cusolver)` ?

Reply:
I moved this part to `_create_U_S_VT`.